### DISCUSSION TITLE
Skip storing to history?

### DISCUSSION THREAD

**Discussion by shripadk** [pmndrs/valtio#450] [Q&A] [answered]
Dear @dai-shi and Valtio contributors,

First of all, I want to thank you for making an awesome library for handling state management. Love the simplicity of Valtio! Has made me really productive! Don't miss Redux at all ever since I moved to using Valtio. Thank you so much!

I am using `proxyWithHistory`. Wanted to know if I could skip saving history for certain operations. I am storing UI state in the store itself and I wish to ignore certain UI state changes (like recording hover over an element). Is there a way to skip storing certain state changes to history?

Would it be better to make storing to history manual (I noticed that there is a `saveHistory` function in the code, maybe can call it explicitly whenever I wish to store to history)? That way we can have higher control over what goes into history.

Or do you suggest I create a separate store for handling ephemeral UI states?

Thanks!

**Comment by shripadk** [comment]
Decided to use a separate store for handling ephemeral UI states. It makes more sense as creating new stores is trivial. Then can only use `proxyWithHistory` specifically for handling history. Just leaving this question here in case anyone else has the same doubt.

**Comment by dai-shi** [comment, marked as answer]
proxyWithHistory covers limited use cases, intentionally. Because there are so many different use cases, we just let users customize as they need.

> I noticed that there is a saveHistory function in the code, maybe can call it explicitly whenever I wish to store to history

Yeah, you can modify saveHistory as you need.
```js
const state = proxyWithHistory(...)
const originalSaveHistory = state.saveHistory
state.saveHistory = () => {
  if (some_criteria) {
    origSaveHistory()
  }
}
```

> Or do you suggest I create a separate store for handling ephemeral UI states?

This sounds better in general. And you did so. 👍
